synapseformation/client.py

```diff
@@ -0,0 +1,87 @@
+"""Synapse Formation client"""
+import synapseclient
+from synapseclient import Synapse
+
+from .create import SynapseCreation
+from . import utils
+
+
+# def expand_config(config: dict) -> dict:
+#     """Expands shortened configuration to the official json format"""
+#     # TODO: Once all resources are either under a key mapping or lists
+#     # this will have to chance
+#     if isinstance(config, dict) and config.get('type') == "Project":
+#         # TODO: figure out how to clean this up.
+#         # Remove yaml anchor objects as they are expanded
+#         to_remove = [key for key in config
+#                      if key not in ["type", "name", "children"]]
+#         for key in to_remove:
+#             config.pop(key)
+#         # Get children if exists
+#         children = config.get('children')
+#         if children is not None:
+#             expand_config(children)
+#     else:
+#         # Loop through folders and create them
+#         for index, folder in enumerate(config):
+#             # The folder configuration can be lists or dicts
+#             # Must pull out children if it exists
+#             if isinstance(folder, dict):
+#                 children = folder.get('children')
+#             else:
+#                 config[index] = {"name": folder,
+#                                  "type": "Folder"}
+#                 children = None
+
+#             # Create nested folders
+#             if children is not None:
+#                 expand_config(children)
+#     return config
+
+
+def _create_synapse_resources(syn: Synapse, config: dict,
```

Comment:
I created new subtasks on JIRA for tracking the creation of a schema so we can validate assumptions on the template structure.